docs/advanced/misc.rst

```diff
@@ -135,22 +135,16 @@ has been executed:
 
 Naturally, both methods will fail when there are cyclic dependencies.
 
-Note that compiling code which has its default symbol visibility set to
-*hidden* (e.g. via the command line flag ``-fvisibility=hidden`` on GCC/Clang) can interfere with the
-ability to access types defined in another extension module. Workarounds
-include changing the global symbol visibility (not recommended, because it will
-lead unnecessarily large binaries) or manually exporting types that are
-accessed by multiple extension modules:
+Note that pybind11 code compiled with hidden-by-default symbol visibility (e.g.
+via the command line flag ``-fvisibility=hidden`` on GCC/Clang), which is
+strongly recommended for pybind11 modules, can interfere with the ability to
```

Comment:
I guess it's required, not just recommended anymore. Maybe simplify to something along the lines of: "By default,  pybind11 modules are compiled with hidden symbol visibility (see :ref:`...`), which can interfere..."